### PR TITLE
DS Migration : Auth pages (Forgot/reset password, Oops page)

### PR DESCRIPTION
**Pull request by HichamELBSI**
Signed-off-by: HichamELBSI <elabbassih@gmail.com>

### What does it do?

Migrate the auth pages (Forgot/reset password, Oops page).

Screenshot 2021-08-18 at 17 39 28
Screenshot 2021-08-18 at 17 39 43
Screenshot 2021-08-18 at 17 40 00

### CODE REVIEW

File: packages/core/admin/admin/src/pages/AuthPage/components/ForgotPassword/index.js [hunk 1]
```diff
@@ -1,53 +1,75 @@
...
+            validationSchema={schema}
+            validateOnChange={false}
+          >
+            {({ values, errors, handleChange }) => (
+              <Form noValidate>
+                <Column>
+                  <Logo />
+                  <Box paddingTop="6" paddingBottom="7">
+                    <H1 id="password-forgotten">
+                      {formatMessage({ id: 'Auth.form.button.password-forgotten' })}
```
Comment (soupette) [inline packages/core/admin/admin/src/pages/AuthPage/components/ForgotPassword/index.js:36]: Can you add the `defaultMessage` for all react-intl text?

File: packages/core/admin/admin/src/pages/AuthPage/components/Register/index.js [hunk 2]
```diff
@@ -33,26 +37,70 @@ const A = styled.a`
...
+        } catch (err) {
+          const errorMessage = get(err, ['response', 'data', 'message'], 'An error occurred');
+
+          toggleNotification({
+            type: 'warning',
+            message: errorMessage,
+          });
+
+          // Redirect to the oops page in case of an invalid token
+          // @alexandrebodin @JAB I am not sure it is the wanted behavior
```
Comment (soupette) [inline packages/core/admin/admin/src/pages/AuthPage/components/Register/index.js:72]: Yes it is!

**Review by soupette — CHANGES_REQUESTED** [review]
Good job!

Can you make sure all translation messages have a `defaultMessage`?
Also, can you add some tests?

File: packages/core/admin/admin/src/pages/AuthPage/components/Register/index.js (hunk 2 above)
Comment (HichamELBSI) [reply packages/core/admin/admin/src/pages/AuthPage/components/Register/index.js:72]: That was an old comment from you 😄 . Are we sure about that?

File: packages/core/admin/admin/src/translations/en.json [hunk 3]
```diff
@@ -34,6 +37,7 @@
   "Auth.form.lastname.label": "Lastname",
   "Auth.form.lastname.placeholder": "Doe",
   "Auth.form.password.label": "Password",
+  "Auth.form.password.hint": "Password must contain at least 8 characters, 1 uppercase, 1 lowercase and 1 number",
```
Comment (derrickmehaffy) [inline packages/core/admin/admin/src/translations/en.json:40]: I think this comment is wrong, although I'm looking for the regex pattern we use. I think a symbol is required as well.

Comment (HichamELBSI) [reply packages/core/admin/admin/src/translations/en.json:40]: Currently, the symbol isn't required for the admin registration form.

File: packages/core/admin/admin/src/pages/AuthPage/components/Login/tests/BaseLogin.test.js [hunk 4]
```diff
@@ -0,0 +1,165 @@
+import React from 'react';
+import { render } from '@testing-library/react';
+import { ThemeProvider } from '@strapi/parts/ThemeProvider';
+import { lightTheme } from '@strapi/parts/themes';
+import { Router } from 'react-router-dom';
+import * as yup from 'yup';
+import { createMemoryHistory } from 'history';
+import BaseLogin from '../BaseLogin';
+
+jest.mock('react-intl', () => {
```
Comment (soupette) [inline packages/core/admin/admin/src/pages/AuthPage/components/Login/tests/BaseLogin.test.js:10]: Why do you need to mock react-intl?

Same for the Form component

File: packages/core/admin/admin/src/pages/AuthPage/components/Register/index.js (hunk 2 above)
Comment (soupette) [reply packages/core/admin/admin/src/pages/AuthPage/components/Register/index.js:72]: Yes

**Review by soupette — APPROVED** [review]
LGTM

**Pull request merged**